### ISSUE TITLE
Implement struct block_list for skill functions

### ISSUE THREAD

**Issue by SombraRO** [HerculesWS/Hercules#2793]
**Is your feature request related to a problem? Please describe.**
None

**Describe the solution you'd like**
None

**Describe alternatives you've considered**
None

**Additional context**
I posted about it when @Kenpachi2k13  was working on cleaning up skills.

The idea is to allow manipulating the functions for different types, like.

- Monsters
- Chars
- Pets...

https://github.com/HerculesWS/Hercules/pull/2731#issuecomment-636410479

https://github.com/HerculesWS/Hercules/pull/2731#issuecomment-636502072

**Comment by SombraRO** [comment]
@Kenpachi2k13 something about that?

**Comment by Kenpachi2k13** [comment]
> @Kenpachi2k13 something about that?

Yes. I talked to @MishimaHaruna and your feature request was accepted.
I'll try to get this done quickly.

**Comment by SombraRO** [comment]
@Kenpachi2k13 news ?

**Comment by SombraRO** [comment]
@Kenpachi2k13 @MishimaHaruna @4144 Any info about that?

**Comment by SombraRO** [comment]
@csnv
@guilherme-gm

**Comment by SombraRO** [comment]
@guilherme-gm @dastgirp 

as Kenpachi2k13 is no longer in Hercules, could you guys look at it ?

I think such a feature would expand the use of the plugins
